client/state/admin-color/selectors/index.js

```diff
@@ -3,3 +3,7 @@ import 'calypso/state/admin-color/init';
 export function getAdminColor( state, siteId ) {
 	return state?.adminColor?.[ siteId ] || null;
 }
+
+export function isFetchingAdminColor( state, siteId ) {
+	return ! state?.adminColor?.[ siteId ];
```

Comment:
Did you consider adding a loading property to the [state](https://github.com/Automattic/wp-calypso/blob/c4d88f57af9784b7c862c9abf0eae72a5f6cffee/client/state/admin-color/reducer.js#L6)?

That would allow us to [handle a request failure](https://github.com/Automattic/wp-calypso/blob/c4d88f57af9784b7c862c9abf0eae72a5f6cffee/client/state/data-layer/wpcom/sites/admin-color/index.js#L24) properly. 

Currently, it seems like if `/sites/${ action.siteId }/admin-color/` fails, we won't render the Masterbar. 

Since color is not critical to our user stories, we probably want to avoid that scenario.

Reply:
Oh, nice catch!

I updated the code to fall back to the `default` scheme, when the request fails while the persisted value is still empty (i.e., during the first login). Note that if the request fails while the persisted value is already available, it won't do anything. This way the masterbar/sidebar will always render in all cases.

I wasn't able to add a "requesting" property cleanly to the state, because somehow the shape of the value in the redux state is just the string value, not an object 🥲 So the alternative is to add another `adminColorRequesting` reducer, which I feel is too much given that this will go away in the future after Untangling Calypso is finished.